internal/server/resolvers/schema.resolvers.go

```
@@ -42,6 +43,43 @@ func (r *queryResolver) Hosts(ctx context.Context) ([]*graphql1.Host, error) {
+// Contract is the resolver for the contract field. Returns nil (not an
+// error) when the id is unknown so consumers can distinguish "missing"
+// from "fetch failed."
+func (r *queryResolver) Contract(ctx context.Context, id string) (*graphql1.Contract, error) {
+	if r.ContractsProvider == nil {
+		return nil, fmt.Errorf("contracts provider not initialised")
+	}
+	key := contracts.ContractID(stripContractIDPrefix(id))
+	c, _, err := r.ContractsProvider.Get(ctx, key)
+	if err != nil {
+		// Unknown id is the common case; surface as nil result.
+		return nil, nil
```

golangci-lint (2.12.1)

[error] error is not nil (line 54) but it returns nil

(nilerr)